Sources/Protocols/MessagesDisplayDelegate.swift

```diff
@@ -45,10 +66,31 @@ public extension TextMessageDisplayDelegate {
 
 }
 
+/// A protocol used by the `MessagesViewController` to customize the appearance of a `MessagesCollectionViewCell`.
 public protocol MessagesDisplayDelegate: class {
 
+    /// Specifies the `MessageStyle` to be used for a `MessageContainerView`.
+    ///
+    /// - Parameters:
+    ///   - message: The `MessageType` that will be displayed by this cell.
+    ///   - indexPath: The `IndexPath` of the cell.
+    ///   - messagesCollectionView: The `MessagesCollectionView` in which this cell will be displayed.
+    ///
+    /// The default value returned by this method is `MessageStyle.bubble`.
     func messageStyle(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> MessageStyle
-    
+
+    /// Specifies the background color of the `MessageContainerView`.
+    ///
+    /// - Parameters:
+    ///   - message: The `MessageType` that will be displayed by this cell.
+    ///   - indexPath: The `IndexPath` of the cell.
+    ///   - messagesCollectionView: The `MessagesCollectionView` in which this cell will be displayed.
+    ///
+    /// The default value is `UIColor.clear` for emoji messages. For all other `MessageData` cases, the color depends on the `Sender`:
+    ///
+    /// Current Sender: Green
+    ///
+    /// All other Senders: Gray
     func backgroundColor(for message: MessageType, at  indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> UIColor
     
     func messageHeaderView(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> MessageHeaderView
```

Comment:
What about this one 😛

Reply:
Yeah, you caught me. I got tired 👍 Going to review all this now that I have more energy